### ISSUE TITLE
Understanding the dimensions for the classifier for SoftmaxLoss

### ISSUE THREAD

**Issue by Aniket-Pradhan** [UKPLab/sentence-transformers#292]
Hello there

Thanks a lot for your work, this has truly brought a lot of insight into my work.

I was making my own (with a little variation) Siamese BERT model and was wondering to use the SoftmaxLoss there. From the code, I am unable to understand the dimensionality of the classifier in the SoftmaxLoss.

The output dimension would be equal to [`num_labels`](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/losses/SoftmaxLoss.py#L30) argument. Say, I have a sentence pair and a label {0, 1}. Therefore, I would put the `num_labels` as 2. Therefore, the output of the Linear layer would be of size `[batch_size, num_labels]`.

However, my target labels would be a tensor of size `[batch_size]`, and hence I should not be able to calculate the cross-entropy loss.

I tried running the model from this library using `num_labels = 2`, and it works. I was unable to find any other place in the source where it converts the labels to dimension 2. Can you help me understand what I am missing here?

**Comment by Aniket-Pradhan** [comment]
Ahh. the cross-entropy loss requires the input to be of size `[batch_size, num_classes]` and the targets of shape `[batch_size]`. Furthermore, the targets should be of dtype long (another reason why my code was crashing).
